explore/src/main/scala/explore/constraints/ConstraintsSummaryTile.scala

```diff
@@ -103,177 +110,195 @@ object ConstraintsSummaryTile:
     )
 
   private case class Body(
-    userId:           Option[User.Id],
-    programId:        Program.Id,
-    constraintList:   ConstraintGroupList,
-    expandedIds:      View[SortedSet[ObsIdSet]],
-    columnVisibility: View[ColumnVisibility]
+    userId:                  Option[User.Id],
+    programId:               Program.Id,
+    constraintList:          ConstraintGroupList,
+    calibrationObservations: Set[Observation.Id],
+    expandedIds:             View[SortedSet[ObsIdSet]],
+    columnVisibility:        View[ColumnVisibility]
   ) extends ReactFnProps(Body.component)
 
   private object Body:
-    private type Props = Body
-
     private val ColDef = ColumnDef[ConstraintGroup]
 
     private val ColumnClasses: Map[ColumnId, Css] = Map(
       EditColumnId -> (ExploreStyles.StickyColumn |+| ExploreStyles.ConstraintsSummaryEdit)
     )
 
     private val component =
-      ScalaFnComponent
-        .withHooks[Props]
-        .useContext(AppContext.ctx)
-        .useMemoBy((_, _) => ()): // Cols never changes, but needs access to props
-          (props, ctx) =>
-            _ =>
-              def column[V](id: ColumnId, accessor: ConstraintGroup => V)
-                : ColumnDef.Single.NoMeta[ConstraintGroup, V] =
-                ColDef(id, accessor, ColumnNames(id))
+      ScalaFnComponent[Body]: props =>
+        for {
+          ctx   <- useContext(AppContext.ctx)
+          cols  <- useMemo(()): // Cols never changes, but needs access to props
```

Comment:
Unrelated to the PR: in other places where we have stable columns definitions, we have moved them to a val with `.reuseAlways`, which declutters the hook section.

Reply:
Oh right, it needs the programId.